extensions/amp-autocomplete/0.1/autocomplete-binding-inline.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ownProperty} from '../../../src/utils/object';
+import {tryFocus} from '../../../src/dom';
+import {userAssert} from '../../../src/log';
+
+/**
+ * Inline implementation of autocomplete. This supports autocompleting
+ * multiple input values as part of a larger freeform input element.
+ * @implements {./autocomplete-binding-def.AutocompleteBindingDef}
+ * @private
+ */
+export class AutocompleteBindingInline {
+  /**
+   * Stores the regex match value associated with the portion of the user input to suggest against.
+   * For use when "inline_" is true.
+   * @param {string} trigger
+   */
+  constructor(trigger) {
+    userAssert(
```

Comment:
As discussed offline, validating the `trigger` value should become a concern for this binding and then this assertion becomes redundant :)